utils/src/treev2/quickPickWizard/compatibility/CompatibilityContextValueQuickPickStep.ts

```diff
@@ -0,0 +1,54 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as types from "../../../../index";
+import { ContextValueFilterQuickPickOptions, ContextValueQuickPickStep } from "../ContextValueQuickPickStep";
+import { getLastNode } from "../QuickPickWizardContext";
+import { AzExtTreeItem } from "../../../tree/AzExtTreeItem";
+import { AzExtParentTreeItem } from "../../../tree/AzExtParentTreeItem";
+import { isWrapper } from "../../../registerCommandWithTreeNodeUnwrapping";
+import { isAzExtParentTreeItem } from "../../../tree/isAzExtParentTreeItem";
+
+/**
+ * Provides compatability with {@link AzExtParentTreeItem.pickTreeItemImpl}
+ */
+export class CompatibilityContextValueQuickPickStep<TContext extends types.QuickPickWizardContext, TOptions extends ContextValueFilterQuickPickOptions> extends ContextValueQuickPickStep<TContext, TOptions> {
+
+    public override async prompt(wizardContext: TContext): Promise<void> {
+        await this.provideCompatabilityWithPickTreeItemImpl(wizardContext) || await super.prompt(wizardContext);
+    }
+
+    /**
+     * Mimics how the legacy {@link AzExtParentTreeItem.pickChildTreeItem}
+     * uses {@link AzExtParentTreeItem.pickTreeItemImpl} to customize the tree item picker.
+     *
+     * An example customization is skipping having to pick a UI-only node (ex: App Settings parent node)
+     */
+    private async provideCompatabilityWithPickTreeItemImpl(wizardContext: TContext): Promise<boolean> {
+        const lastPickedItem = getLastNode(wizardContext);
+        const lastPickedItemUnwrapped = isWrapper(lastPickedItem) ? lastPickedItem.unwrap() : lastPickedItem;
+        if (isAzExtParentTreeItem(lastPickedItemUnwrapped)) {
+            const children = await this.treeDataProvider.getChildren(lastPickedItem);
+            if (children && children.length) {
+                const customChild = await this.getCustomChildren(wizardContext, lastPickedItemUnwrapped);
+
+                const customPick = children.find((child) => {
+                    const ti: AzExtTreeItem = isWrapper(child) ? child.unwrap() : child as unknown as AzExtTreeItem;
+                    return ti.fullId === customChild?.fullId;
+                });
+
+                if (customPick) {
+                    wizardContext.pickedNodes.push(customPick);
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    private async getCustomChildren(context: TContext, node: AzExtParentTreeItem): Promise<AzExtTreeItem | undefined> {
```

Comment:
Shouldn't we also filter based on `exclude`?

Reply:
We're only passing `include` because that's what `AzExtParentTreeItem.pickTreeItemImpl` is expecting. We've never had an `exclude` option before, so it will work as expected by ignoring it. Only v2 extensions will use the `exclude` property, and they won't be using the compat steps.